python/ray/ml/predictor.py

```diff
@@ -51,3 +52,93 @@ def __reduce__(self):
             "to serialize a checkpoint and initialize the Predictor with "
             "Predictor.from_checkpoint."
         )
+
+
+class BatchPredictor(Predictor):
+    """Batch predictor class.
+
+    Takes a predictor class and a checkpoint and provides an interface to run
+    batch scoring on Ray datasets.
+
+    This batch predictor wraps around a predictor class and executes it
+    in a distributed way when calling ``predict()``.
+
+    Attributes:
+        checkpoint: Checkpoint loaded by the distributed predictor objects.
+        predictor_cls: Predictor class reference. When scoring, each scoring worker
+            will create an instance of this class and call ``predict(batch)`` on it.
+        **predictor_kwargs: Keyword arguments passed to the predictor on
+            initialization.
+
+    """
+
+    def __init__(
+        self, checkpoint: Checkpoint, predictor_cls: Type[Predictor], **predictor_kwargs
+    ):
+        # Store as object ref so we only serialize it once for all map workers
+        self.checkpoint_ref = checkpoint.to_object_ref()
+        self.predictor_cls = predictor_cls
+        self.predictor_kwargs = predictor_kwargs
+
+    @classmethod
+    def from_checkpoint(
+        cls, checkpoint: Checkpoint, predictor_cls: Type[Predictor], **kwargs
+    ) -> "BatchPredictor":
+        return BatchPredictor(
+            checkpoint=checkpoint, predictor_cls=predictor_cls, **kwargs
+        )
+
+    def predict(
+        self,
+        data: ray.data.Dataset,
+        *,
+        batch_size: int = 4096,
+        max_scoring_actors: Optional[int] = None,
```

Comment:
Could we also add min_scoring_actors, so that the user can setup a fixed size pool if needed?

Reply:
at this point we may as well pass in a scaling config 🤔